### ISSUE TITLE
WebSocket generates System.TimeoutException

### ISSUE THREAD

**Issue by aaronzper** [RogueException/Discord.Net#1220]
When running a Discord.Net bot in Visual Studio Community for Mac, the initial connection to Discord with the WebSocket times out. After disconnecting, it attempts to reconnect to Discord, and the same exception is thrown.

The log is as follows:
```[INFO] 11:32:47 Discord     Discord.Net v2.0.0-beta (API v6)
[INFO] 11:32:49 Gateway     Connecting
[INFO] 11:33:19 Gateway     System.TimeoutException: The operation has timed out.
  at Discord.ConnectionManager.WaitAsync () [0x00070] in <972b1e6a8d8c488eb7d3a2d2ae616beb>:0 
  at Discord.WebSocket.DiscordSocketClient.OnConnectingAsync () [0x0040d] in <972b1e6a8d8c488eb7d3a2d2ae616beb>:0 
  at Discord.WebSocket.DiscordSocketClient.OnConnectingAsync () [0x00591] in <972b1e6a8d8c488eb7d3a2d2ae616beb>:0 
  at Discord.ConnectionManager.ConnectAsync (System.Threading.CancellationTokenSource reconnectCancelToken) [0x002fc] in <972b1e6a8d8c488eb7d3a2d2ae616beb>:0 
  at Discord.ConnectionManager+<>c__DisplayClass28_0.<StartAsync>b__0 () [0x000ce] in <972b1e6a8d8c488eb7d3a2d2ae616beb>:0 
[INFO] 11:33:19 Gateway     Disconnecting
[INFO] 11:33:19 Gateway     Disconnected

**Comment by sasaravi** [comment]
Hey Dude, did you solve the problem? I have the same problem in my mac.

**Comment by Lymdun** [comment]
Same issue here with Linux Debian

**Comment by SubZero0** [comment]
I'll be closing this since a lot of changes were made since it was created.
If you have any issue like this one, I recommend updating to the latest nightly and if not resolved, create a new issue or check #960
